### PR TITLE
Security Chapter

### PR DESCRIPTION
**Pull request by adrianschroeter**

### CODE REVIEW

File: xml/obs_ag_security_concepts.xml [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+    <title>Worker network</title>
+    <para>It is recommended to run the &obs; workers in an isolated network. This is
+          an additional security mechanism in case of a security breach on a worker.
+          This network needs access to the source and repository servers of the &obs;
+          backend, but nowhere else.</para>
+   </sect3>
+   <sect3 xml:id="signd_network">
+    <title>Signer network</title>
+    <para>It is recommended to run the signd on an isolated host. The signer services
+          need to stay on the &obs; backend servers, they are just used for scheduling
```
Comment (tomschr) [inline xml/obs_ag_security_concepts.xml:80]: Removed filler word:

```suggestion
          need to stay on the &obs; backend servers, they are used for scheduling
```

File: xml/obs_ag_security_concepts.xml [hunk 2]
```diff
@@ -0,0 +1,140 @@
...
+          sign jobs. The signd is the critical instance which hosts the single instance 
+          signature key. All further keys which belong to any project are created
+          by the signd, crypted with the instance key and delivered to the backend
```
Comment (tomschr) [inline xml/obs_ag_security_concepts.xml:83]:
```suggestion
          by the signd, encrypted with the instance key and delivered to the backend
```

File: xml/obs_ag_security_concepts.xml [hunk 3]
```diff
@@ -0,0 +1,140 @@
...
+          servers. That way the signd instance is stateless after initial setup 
+          and it is enough to backup the backend servers.
+          Any sign job sends the crypted private key to the signd server which is decrypting
```
Comment (tomschr) [inline xml/obs_ag_security_concepts.xml:86]:
```suggestion
          Any sign job sends the encrypted private key to the signd server which is decrypting
```

File: xml/obs_ag_security_concepts.xml [hunk 4]
```diff
@@ -0,0 +1,140 @@
...
+         on VM layer. The only supported VM types, which are considered as secure are
+         KVM and XEN VMs. The VMs prohibit any network access from the running instances.
+         The build script is always creating a new file system, copies in all required
+         data and executes the VM.</para>
+   <para>The build results get extracted directly from the
+         block device. This is done to avoid mounting it which may can be used to
+         breach the host kernel in the file system code by crafting the used
+         filesystem during build.
+         </para>
+   <para>Every build also stores the used binaries inside of the _buildenv file. This
```
Comment (tomschr) [inline xml/obs_ag_security_concepts.xml:105]:
```suggestion
   <para>Every build also stores the used binaries inside of the <filename>_buildenv</filename> file. This
```

File: xml/obs_ag_security_concepts.xml [hunk 5]
```diff
@@ -0,0 +1,140 @@
...
+         can be used to re-create the build environemnt later, even when newer
+         updates got release meanwhile. This requires however a maintenance setup
+         to avoid that former releases get removed.
+         </para>
+   <para>Note: The build tools (like rpmbuild) are runing usually as non-root user, but 
+         this is not a security instrument. It is only a quality mechanic to ensure that 
+         source packages are rebuildable without root permissions.
+         </para>
+  </sect2>
```
Comment (tomschr) [inline xml/obs_ag_security_concepts.xml:114]: If this is important, use a `<note>` tag. If not, remove the "Note:" word.

File: xml/obs_ag_security_concepts.xml [hunk 6]
```diff
@@ -0,0 +1,140 @@
...
+  <sect2 xml:id="source_revision_system">
+   <title>Source Revision System</title>
+   <para>The source revision storage system is part of &obs;. The identification of sources
+         still happens using MD5 sums for historic reasons. MD5 is considered to be still
+         good enough for identifing a source, but it is known to be attackable. Recent
+         versions of the &osc; command line tooling is sending therefore also a SHA256 sum
+         in addition which is used to detect collisions by the source server.
+         </para>
+   <para>Sources can reference other package sources via _link files. These can be pinned
```
Comment (tomschr) [inline xml/obs_ag_security_concepts.xml:123]:
```suggestion
   <para>Sources can reference other package sources via <filename>_link</filename> files. These can be pinned
```

File: xml/obs_ag_security_concepts.xml [hunk 7]
```diff
@@ -0,0 +1,140 @@
...
+         to a revision or always using the latest one. That way underling changes
+         get merged automatically. This is useful for shared development and automated
+         builds, but it should be avoided for base projects.
+         </para>
+  </sect2>
+  <sect2 xml:id="permission_handling">
+   <title>Permission handling</title>
```
Comment (tomschr) [inline xml/obs_ag_security_concepts.xml:130]:
```suggestion
   <title>Permission Handling</title>
```

**Comment by sknorr** [comment]
Ftr, @adrianschroeter are you converting PRs to drafts make us hold off on reviewing for the moment? Or should we review the draft PRs anyway?

**Comment by adrianschroeter** [comment]
@sknorr I consider draft's as reviewable, but they may should not be merged yet (I know it is not complete yet, but I like to share with others alreads). Makes this sense?

**Comment by sknorr** [comment]
Makes sense, thanks!

**Pull request merged**